commons/src/main/java/fr/free/nrw/commons/upload/MwVolleyApi.java

```diff
@@ -0,0 +1,161 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+import android.util.Log;
+
+import com.android.volley.Cache;
+import com.android.volley.NetworkResponse;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.VolleyError;
+import com.android.volley.toolbox.HttpHeaderParser;
+import com.android.volley.toolbox.JsonRequest;
+import com.android.volley.toolbox.Volley;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+
+import java.io.UnsupportedEncodingException;
+
+public class MwVolleyApi {
+
+    private static RequestQueue REQUEST_QUEUE;
+    private static final Gson GSON = new GsonBuilder().create();
+    private Context context;
+
+    public MwVolleyApi(Context context) {
+        this.context = context;
+    }
+    public void request(String apiUrl) {
+        JsonRequest<QueryResponse> request = new QueryRequest(apiUrl,
+                new LogResponseListener<QueryResponse>(), new LogResponseErrorListener());
+        getQueue().add(request);
+    }
+
+    private RequestQueue getQueue() {
+        return getQueue(context);
+    }
+
+    private static RequestQueue getQueue(Context context) {
+        if (REQUEST_QUEUE == null) {
+            REQUEST_QUEUE = Volley.newRequestQueue(context.getApplicationContext());
+        }
+        return REQUEST_QUEUE;
+    }
+
+    private static class LogResponseListener<T> implements Response.Listener<T> {
```

Comment:
Yeah, but we will use the categories extracted in the next step (Display the resulting categories to the user alongside the "previous categories" that are currently already displayed)